### PR TITLE
Upstream animation-ready-promise.html from Blink

### PR DESCRIPTION
**Pull request by suzyh**

### CODE REVIEW

File: web-animations/interfaces/Animation/ready.html [hunk 1]
```diff
@@ -93,5 +93,89 @@
...
+  animation.play();
+  assert_not_equals(animation.ready, promise);
+}, 'The ready promise should be replaced when the animation enters the pending state');
+
+async_test(function(t) {
+  var animation = createDiv(t).animate(null, 100 * MS_PER_SEC);
+  animation.ready.then(function() {
+    t.step(function() {
+      assert_unreached();
+    });
```
Comment (alancutter) [inline web-animations/interfaces/Animation/ready.html:129]: t.unreached_func(description)

File: web-animations/interfaces/Animation/ready.html [hunk 2]
```diff
@@ -93,5 +93,89 @@
...
+async_test(function(t) {
+  var animation = createDiv(t).animate(null, 100 * MS_PER_SEC);
+  animation.cancel();
+  animation.ready.then(function(p) {
+    t.step(function() {
+      assert_equals(p, animation);
+    });
+    t.done();
+  });
+}, 'The ready promise should be resolved when an animation is in the idle play state');
```
Comment (alancutter) [inline web-animations/interfaces/Animation/ready.html:107]: promise_test would reduce the number of lines here by 4.

File: web-animations/interfaces/Animation/ready.html [hunk 3]
```diff
@@ -93,5 +93,89 @@
...
+    });
+    t.done();
+  });
+}, 'A pending ready promise should be resolved and not replaced when the animation enters the running state');
+
+async_test(function(t) {
+  var animation = createDiv(t).animate(null, 100 * MS_PER_SEC);
+  var promise = animation.ready;
+  animation.finish();
+  promise.then(function(p) {
```
Comment (alancutter) [inline web-animations/interfaces/Animation/ready.html]: Shouldn't this be animation.finished?

File: web-animations/interfaces/Animation/ready.html [hunk 4]
```diff
@@ -93,5 +93,89 @@
...
+  var promise = animation.ready;
+  animation.pause();
+  promise.then(function(p) {
+    t.step(function() {
+      assert_equals(p, animation);
+      assert_equals(animation.ready, promise);
+    });
+    t.done();
+  });
+}, 'A pending ready promise should be resolved and not replaced when the animation enters the paused state');
```
Comment (alancutter) [inline web-animations/interfaces/Animation/ready.html:174]: Ditto async_test -> promise_test comment for the above three tests.

File: web-animations/interfaces/Animation/ready.html (hunk 1 above)
Comment (suzyh) [inline web-animations/interfaces/Animation/ready.html:129]: Done

File: web-animations/interfaces/Animation/ready.html (hunk 2 above)
Comment (suzyh) [inline web-animations/interfaces/Animation/ready.html:107]: Done

File: web-animations/interfaces/Animation/ready.html (hunk 3 above)
Comment (suzyh) [inline web-animations/interfaces/Animation/ready.html]: Done

File: web-animations/interfaces/Animation/ready.html (hunk 4 above)
Comment (suzyh) [inline web-animations/interfaces/Animation/ready.html:174]: Done

**Comment by alancutter** [comment]
lgtm

**Pull request merged**